File: C++/Level.3/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

int N;
int cnt;
vector<int> visited;
vector<vector<int>> network;

void DFSUtil(int index) {
    if (visited[index]) {
        return;
    }
    visited[index] = 1;
    for (int i = 0; i < N; ++i) {
        if (network[index][i] == 1) {
            DFSUtil(i);
        }
    }
}

int DFS(int n, vector<vector<int>> computers) {
    N = n;
	cnt = 0;
    network = computers;
    visited.assign(N, 0);
    
    for (int i = 0; i < N; ++i) {
        if (!visited[i]) {
            ++cnt;
            DFSUtil(i);
        }
    }
    return cnt;
}

int solution(int n, vector<vector<int>> computers) {
    return DFS(n, computers);
}

File: JavaScript/Level.3/네트워크.js
function solution(n, computers) {
    let cnt = 0;
    let visited = Array.from({ length: n }, () => 0);
    function DFSUtil(vertex) {
        for (let i = 0; i < n; ++i) {
            if (computers[vertex][i] === 1 && !visited[i]) {
                visited[i] = 1;
                DFSUtil(i);
            }
        }
    }
    for (let i = 0; i < n; ++i) {
        if (!visited[i]) {
            ++cnt;
            visited[i] = 1;
            DFSUtil(i);
        }
    }
    return cnt;
}
